### PR TITLE
Create simple GitHub action for lint/build/test.

### PR DESCRIPTION
**Pull request by OmarDesoky**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
```
Comment (phella) [inline .github/workflows/go.yml]: nit: go version 1.19

**Comment by phella** [comment]
I see it is failing in limiting step can we remove it for now?

**Comment by phella** [comment]
nice pr ❤️🧨

**Review by phella — APPROVED** [review]

**Comment by OmarDesoky** [comment]
> I see it is failing in limiting step can we remove it for now?

Let's start enforcing it, so we have clean main/master.

**Pull request merged**